namespace test.features;
using test.features.Fraction;

println("Testing handling user defined exceptions: ");
try {
    val frac = Fraction(3, 0);
    println("Try statement executed successfully.");
}
catch(RuntimeException e) { 
    println("Caught " + e.getClassName() + " with message: " + e.message);
}
finally { 
    println("Finally clean up...");
}
println("");

println("Testing handling native exception: ");
try { 
    val range = 3.."abc";
}
catch(IllegalArgumentException e){ 
    println("Caught " + e.getClassName() + " with message: " + e.message);
}
finally { 
    println("Finally clean up...");
}
println("");